### DISCUSSION TITLE
USBDONGLE_WB55 machine.lightsleep hangs board forever

### DISCUSSION THREAD

**Discussion by MJank0wski** [micropython/micropython#13535] [STM32 / Pyboard]
Hello,
This is my first post here so I would like to say hello to all and thanks for your involvement in MicroPython development.

The issue that I have is related with machine.lightsleep function run on stm32wb55 (USBDONGLE_WB55) with MicroPython v1.22.1 on 2024-01-05; USBDongle-WB55 with STM32WB55CGU6.

Using REPL if I run
```py
import machine

machine.lightsleep(1000)
```
the REPL becomes inactive for forever.

machine.lightsleep(1) executes immediately and REPL is active.

machine.deepsleep(10000) makes the board reset after 10 sec so it works correctly.

Is it possible to wake up from machine.lightsleep using RTC?

The STM32WB55CGU6 supports several power modes, I would like to use:
1. Shutdown with RTC (it seems that this is run by machine.deepsleep)
2. Stop 2 with Advertising  - is it possible to enter to this mode using MicroPython?

**Comment by andrewleech** [comment]
This sounds like it could be an instance of this issue: https://github.com/micropython/micropython/pull/8304

The board might be waking up fine, but the USB has been knocked out so you never see repl again.

**Reply by MJank0wski** [reply]
It seems that the board does not wake up. 
If I place the following code to the main.py:

```py
# main.py -- put your code here!
import pyb
import machine
from pyb import Switch

pyb.LED(2).off()

def lightSlpEnter():
    pyb.LED(2).toggle()
    machine.lightsleep(5000)
    pyb.LED(2).toggle()
    
sw=Switch()
sw.callback(lightSlpEnter)
```
The board reacts on button press only once. The green led is turned on and it stays in this state forever. There is no reaction on the next button clicks.

Maybe there is something wrong with the above code?
